doozer/doozerlib/cli/scan_osh.py

```diff
@@ -0,0 +1,157 @@
+import click
+import koji
+from doozerlib.cli import cli, click_coroutine, pass_runtime
+from doozerlib.runtime import Runtime
+from doozerlib.exectools import fire_and_forget, cmd_gather
+from doozerlib.util import cprint
+from typing import Optional
+
+
+class ScanOshCli:
+    def __init__(self, runtime: Runtime, last_brew_event: int, dry_run: bool, nvrs: Optional[list],
+                 check_triggered: Optional[bool]):
+        self.runtime = runtime
+        self.last_brew_event = last_brew_event
+        self.dry_run = dry_run
+        self.brew_tags = []
+        self.specific_nvrs = nvrs
+        self.check_triggered = check_triggered
+        self.nvrs = []
+
+        # Initialize runtime and brewhub session
+        self.runtime.initialize(clone_distgits=False)
+        self.koji_session = koji.ClientSession(self.runtime.group_config.urls.brewhub)
+
+    async def untriggered_nvrs(self):
+        """
+        So we might have already triggered a scan for the same NVR. If this flag is enabled, we want to check
+        if it has and not re-trigger it again.
+        But please note that this will take a lot of time since it has to run for all 200+ packages.
+        """
+        untriggered_nvrs = []
+        for nvr in self.nvrs:
+            rc, _, _ = cmd_gather(f"osh-cli find-tasks --nvr {nvr}")
+
+            if rc == 1:
+                # Get the list of NVRS that a task hasn't already been triggered for
+                untriggered_nvrs.append(nvr)
+
+        self.nvrs = untriggered_nvrs
+
+    def get_tagged_latest(self, tag):
+        """
+        Returns the latest RPMs and builds tagged in to the candidate tag received as input
+        """
+        latest_tagged = self.koji_session.listTagged(tag=tag, latest=True)
+        if latest_tagged:
+            return latest_tagged
+        else:
+            return []
+
+    def get_tagged_all(self, tag):
+        """
+        Returns all the RPMs and builds that are currently in the candidate tag received as input
+        """
+        latest_tagged = self.koji_session.listTagged(tag=tag, latest=False)
+        if latest_tagged:
+            return latest_tagged
+        else:
+            return []
+
+    async def trigger_scan(self):
```

Comment:
Does not look like a coroutine, it does not `await` anything. Can be a regular sync function

Reply:
https://github.com/openshift-eng/art-tools/pull/51/commits/7c715b307aa02efa5d521ca0e931c9a2ae18766e